keel-api/keel-api.gradle.kts

```diff
@@ -12,6 +12,7 @@ plugins {
 dependencies {
   api("com.netflix.spinnaker.kork:kork-plugins-api:${property("korkVersion")}")
   api("com.netflix.spinnaker.kork:kork-exceptions:${property("korkVersion")}")
+  api("com.netflix.spinnaker.kork:kork-artifacts:${property("korkVersion")}")
```

Comment:
Required for the `Artifact` model used in `ArtifactEvent`. Unsure whether it's worth pulling the entire library or if we should just create a duplicate in keel.

Reply:
We can leave this as-is, but it's definitely not going to fly long-term. `kork-artifacts` brings in way too many transitive dependencies: https://github.com/spinnaker/kork/blob/master/kork-artifacts/kork-artifacts.gradle#L8-L9

Both of those deps are destined to ruin days.

Reply:
I think we should pull in this dependency, because it's how artifacts are represented in all of spinnaker. we don't want to miss updates to it and find out later we're breaking something I think.

Reply:
My concern was that it may break the requirements for Spinnaker plugins. @robzienert @jonsie thoughts?

Reply:
The way the module is written today is a definite no-go long-term. I agree that we should use those models, but the Jackson and Jinjava dependencies have got to go.

Reply:
I'm inclined to create a dupe of that one class we need in keel because it's not like we use shared model libraries for anything else anyway... @emjburns How do you feel about that?

Reply:
Thinking out loud here -- I think as a follow up item to this PR we could pull `Artifact` out to new module called `kork-api` that is basically vanilla Java + lombok.  We could then provide some Jackson Mixin classes to be used when configuring ObjectMapper to instruct Jackson how to handle `Artifact` fields -- not totally sure where we would do that configuration though.

Reply:
TBH, I didn’t realize `Artifact` even had Jackson annotations. I assumed it was just a POJO. At this point, I really think we should create a mirror class in keel, so we can remove the dependency on `kork-artifacts` and decouple the two conversations. If/when kork is refactored to be more plugin-friendly, then we can reintroduce the shared class. 

@emjburns do you object?

Reply:
That sounds like the right approach. I'm concerned we might end up accidentally relying on some of those transitive dependencies.

Reply:
didn't realize kork-artifact had so many dependencies, and also didn't see @robzienert's comment when I commented (thanks github, why u so laggy?). Fine with creating a mirror class in keel, as long as it's quite flexible and doesn't barf when properties are added.

Reply:
Replaced use of kork's `Artifact` with a copycat called `SpinnakerArtifact` in keel-api.